src/Types/DescribedList.cs

```diff
@@ -55,32 +63,376 @@ public bool HasField(int index)
         public void ResetField(int index)
         {
             this.CheckFieldIndex(index);
-            this.fields[index] = null;
+            this.fields &= ~(1 << index);
```

Comment:
I would add some comments describing what's actually happening here, as it is a bit tricky to reason about. Especially at first glance.